crates/oxc_linter/src/rules/jsx_a11y/no_aria_hidden_on_focusable.rs

```diff
@@ -0,0 +1,180 @@
+        if let Some(aria_hidden_prop) = has_jsx_prop_lowercase(jsx_el, "aria-hidden") {
+            if is_aria_hidden_true(aria_hidden_prop) && is_focusable(jsx_el) {
+                if let JSXAttributeItem::Attribute(boxed_attr) = aria_hidden_prop {
+                    ctx.diagnostic(NoAriaHiddenOnFocusableDiagnostic(boxed_attr.span));
+                }
+            }
+        }
+    }
+}
+
+fn is_aria_hidden_true(attr: &JSXAttributeItem) -> bool {
+    match attr {
+        JSXAttributeItem::Attribute(attr) => match &attr.value {
+            Some(JSXAttributeValue::StringLiteral(s)) if s.value == "true" => true,
+            None => true,
+            _ => false,
+        },
+        _ => false,
+    }
+}
+
+/// Determines if a JSX element is focusable.
+///
+/// According to the [W3C's DOM Level 2 HTML specification](https://www.w3.org/TR/DOM-Level-2-HTML/html.html), the elements that are focusable are:
+/// - `<a>`, `<area>` with an `href` attribute
+/// - `<button>`, `<input>`, `<select>`, `<textarea>` unless they are disabled
+/// - Any element with a `tabIndex` of zero or greater
+///
+/// This function checks the passed `JSXOpeningElement` against these criteria to determine
+/// if it is focusable.
+///
+/// # Arguments
+///
+/// * `element` - A reference to the JSXOpeningElement to check
+///
+/// # Returns
+///
+/// `true` if the element is focusable, `false` otherwise.
+fn is_focusable(element: &JSXOpeningElement) -> bool {
+    let tag_name = match &element.name {
+        JSXElementName::Identifier(JSXIdentifier { name, .. }) => name.as_str(),
+        _ => return false,
+    };
+
+    let has_tab_index = element.attributes.iter().any(|attr| match attr {
+        JSXAttributeItem::Attribute(attr) => match &attr.name {
+            JSXAttributeName::Identifier(JSXIdentifier { name, .. }) => {
+                name.as_str() == "tabIndex" || name.as_str() == "tabindex"
+            }
+            _ => false,
+        },
+        _ => false,
+    });
+
+    if has_tab_index {
+        return element.attributes.iter().any(|attr| match attr {
+            JSXAttributeItem::Attribute(attr) => match &attr.name {
+                JSXAttributeName::Identifier(JSXIdentifier { name, .. })
+                    if (name.as_str() == "tabIndex" || name.as_str() == "tabindex") =>
+                {
+                    match &attr.value {
+                        Some(JSXAttributeValue::StringLiteral(s)) => {
+                            s.value.parse::<i32>().ok().filter(|&num| num >= 0).is_some()
+                        }
+                        Some(JSXAttributeValue::ExpressionContainer(JSXExpressionContainer {
+                            expression: JSXExpression::Expression(expr),
+                            ..
+                        })) => match expr {
+                            Expression::NumberLiteral(num) => num.value as i32 >= 0,
+                            _ => false,
+                        },
```

Comment:
can we rewrite this to improve perf? 

it looks like we loop over the attributes to check whether there is `tabindex` property.

then loop again to check the value of the prop.

shall we do:
```
if let Some(tab_index_prop) = has_jsx_prop_lowercase(element) {
// check the tab index

}
```

you might also be able to reuse:

https://github.com/oxc-project/oxc/blob/cf0793b67524a61e7b302ec8f21131282bf457c2/crates/oxc_linter/src/rules/jsx_a11y/tab_index_no_positive.rs#L66-L82

Reply:
you are absolutely right. I've refactored the code to use `has_jsx_prop_lowercase` and to reduce attribute iterations!  https://github.com/oxc-project/oxc/pull/1795/commits/1da727200c31cb1f45751aeccd5ff5eb95176700